programs/leap-util/actions/snapshot.cpp

```diff
@@ -10,29 +16,101 @@
 #include <boost/filesystem.hpp>
 #include <boost/filesystem/path.hpp>
 
+using namespace eosio;
+using namespace eosio::chain;
+
+namespace bfs = boost::filesystem;
+
 void snapshot_actions::setup(CLI::App& app) {
-   auto* sub = app.add_subcommand("snapshot", "snapshot utility");
-   sub->add_subcommand("to-json", "convert snapshot file to convert to json format");
+   auto* sub = app.add_subcommand("snapshot", "Snapshot utility");
+   sub->require_subcommand(1);
 
-   // options
-   sub->add_option("--input-file,-i", opt->input_file, "snapshot file to convert to json format, writes to <file>.json if output file not specified (tmp state dir used), and exit.")->required();
-   sub->add_option("--output-file,-o", opt->output_file, "the file to write the output to (absolute or relative path).  if not specified then output is to stdout.");
+   // subcommand -convert snapshot to json
+   auto to_json = sub->add_subcommand("to-json", "Convert snapshot file to json format");
+   to_json->add_option("--input-file,-i", opt->input_file, "Snapshot file to convert to json format, writes to <file>.json if output file not specified (tmp state dir used).")->required();
+   to_json->add_option("--output-file,-o", opt->output_file, "The file to write the output to (absolute or relative path).  If not specified then output is to <input-file>.json.");
+   to_json->add_option("--chain-id", opt->chain_id, "Specify a chain id in case it is not included in a snapshot (for legacy snapshots).");
```

Comment:
If `chain-id` is specified I think it should override the chainid of the snapshot even if the snapshot has one.